filebeat/docs/inputs/input-journald.asciidoc

```diff
@@ -124,36 +111,26 @@ The position to start reading the journal from. Valid settings are:
 {beatname_uc} resends all log messages in the journal.
 * `tail`: Starts reading at the end of the journal. This means that no events
 will be sent until a new message is written.
-* `cursor`: On first read, starts reading at the beginning of the journal. After
-a reload or restart, continues reading at the last known position.
 * `since`: Use the `since` option to determine where to start reading from.
 
-If you have old log files and want to skip lines, start {beatname_uc} with
-`seek: tail` specified. Then stop {beatname_uc}, set `seek: cursor`, and restart
-{beatname_uc}.
+Regardless of the value of `seek` if {beatname_uc} has a state (cursor) for this
+input, the `seek` value is ignored and the current cursor is used. To reset
+the cursor, just change the `id` of the input, this will start from a fresh state.
```

Comment:
This leaves the old registry entries around forever doesn't it?

Reply:
It does, but that is already the normal behaviour if input ID is changed. The only thing that is new is we mentioning it in the docs as a way to ingest things from the beginning.

The current implementation has got a very complicated implementation with seek mode and a fallback seek mode that requires users to stop Filebeat, change the configuration and start again. Which I believe is also not ideal.

Once the registry goes to a checkpoint, there will be only a single entry left by journald input ID, which is very little.

I ended up adding this detail to the documentation because differently from a file that can be copied to be re-ingested, the system journal will always have the same identifier.

At some point (9.0?) we could have a registry clean up that removes entries from any input that cannot be found in the configuration. The way Elastic-Agent configures Filebeat adds an interesting dimension to this problem.